ui/components/ui/box/box.test.js

```diff
@@ -0,0 +1,724 @@
+        'box--md:background-color-info-default',
+      );
+      expect(getByText('Box content')).toHaveClass(
+        'box--lg:background-color-warning-default',
+      );
+    });
+  });
+  describe('width, height', () => {
+    it('should render the Box with the width class', () => {
+      const { getByText } = render(
+        <>
+          <Box width={BLOCK_SIZES.HALF}>Box half</Box>
+          <Box width={BLOCK_SIZES.ONE_FOURTH}>Box one fourth</Box>
+          <Box width={BLOCK_SIZES.MAX}>Box max</Box>
+          <Box width={BLOCK_SIZES.MIN}>Box min</Box>
+        </>,
+      );
+      expect(getByText('Box half')).toHaveClass('box--width-1/2');
+      expect(getByText('Box one fourth')).toHaveClass('box--width-1/4');
+      expect(getByText('Box max')).toHaveClass('box--width-max');
+      expect(getByText('Box min')).toHaveClass('box--width-min');
+    });
+    it('should render the Box with the responsive width classes', () => {
+      const { getByText } = render(
+        <>
+          <Box
+            width={[
+              BLOCK_SIZES.HALF,
+              BLOCK_SIZES.ONE_FOURTH,
+              BLOCK_SIZES.MAX,
+              BLOCK_SIZES.MIN,
+            ]}
+          >
+            Box content
+          </Box>
+        </>,
+      );
+      expect(getByText('Box content')).toHaveClass('box--width-1/2');
+      expect(getByText('Box content')).toHaveClass('box--sm:width-1/4');
+      expect(getByText('Box content')).toHaveClass('box--md:width-max');
+      expect(getByText('Box content')).toHaveClass('box--lg:width-min');
+    });
+    it('should render the Box with the height class', () => {
+      const { getByText } = render(
+        <>
+          <Box height={BLOCK_SIZES.HALF}>Box half</Box>
+          <Box height={BLOCK_SIZES.ONE_FOURTH}>Box one fourth</Box>
+          <Box height={BLOCK_SIZES.MAX}>Box max</Box>
+          <Box height={BLOCK_SIZES.MIN}>Box min</Box>
+        </>,
+      );
+      expect(getByText('Box half')).toHaveClass('box--height-1/2');
+      expect(getByText('Box one fourth')).toHaveClass('box--height-1/4');
+      expect(getByText('Box max')).toHaveClass('box--height-max');
+      expect(getByText('Box min')).toHaveClass('box--height-min');
+    });
+    it('should render the Box with the responsive height classes', () => {
+      const { getByText } = render(
+        <>
+          <Box
+            height={[
+              BLOCK_SIZES.HALF,
+              BLOCK_SIZES.ONE_FOURTH,
+              BLOCK_SIZES.MAX,
+              BLOCK_SIZES.MIN,
+            ]}
+          >
+            Box content
+          </Box>
+        </>,
+      );
+      expect(getByText('Box content')).toHaveClass('box--height-1/2');
+      expect(getByText('Box content')).toHaveClass('box--sm:height-1/4');
+      expect(getByText('Box content')).toHaveClass('box--md:height-max');
+      expect(getByText('Box content')).toHaveClass('box--lg:height-min');
+    });
+  });
+});
```

Comment:
Adding tests. Previously there were none! Tests each prop for singular and responsive array prop values.